src/providers/ogr/qgsogrsourceselect.h

```diff
@@ -0,0 +1,107 @@
+
+#include <ui_qgsogrsourceselectbase.h>
+#include <QDialog>
+#include "qgshelp.h"
+#include "qgsproviderregistry.h"
+#include "qgsabstractdatasourcewidget.h"
+#include "qgis_gui.h"
+
+#define SIP_NO_FILE
+
+/**
+ *  Class for a  dialog to select the type and source for ogr vectors, supports
+ *  file, database, directory and protocol sources.
+ *  \note not available in Python bindings
+ */
+class GUI_EXPORT QgsOgrSourceSelect : public QgsAbstractDataSourceWidget, private Ui::QgsOgrSourceSelectBase
+{
+    Q_OBJECT
+
+  public:
+    QgsOgrSourceSelect( QWidget *parent = nullptr, Qt::WindowFlags fl = 0, QgsProviderRegistry::WidgetMode widgetMode = QgsProviderRegistry::WidgetMode::None );
+    ~QgsOgrSourceSelect();
+    //! Opens a dialog to select a file datasource*/
+    QStringList openFile();
+    //! Opens a dialog to select a directory datasource*/
+    QString openDirectory();
+    //! Returns a list of selected datasources*/
+    QStringList dataSources();
+    //! Returns the encoding selected for user*/
+    QString encoding();
+    //! Returns the connection type
+    QString dataSourceType();
+  private:
+    //! Stores the file vector filters */
+    QString mVectorFileFilter;
+    //! Stores the selected datasources */
+    QStringList mDataSources;
+    //! Stores the user selected encoding
+    QString mEnc;
+    //! Stores the datasource type
+    QString mDataSourceType;
+    //! Embedded dialog (do not call parent's accept) and emit signals
+    QgsProviderRegistry::WidgetMode mWidgetMode = QgsProviderRegistry::WidgetMode::None;
+
+  public slots:
+    void addButtonClicked() override;
+
+  private slots:
+    //! Opens the create connection dialog to build a new connection
+    void addNewConnection();
+    //! Opens a dialog to edit an existing connection
+    void editConnection();
+    //! Deletes the selected connection
+    void deleteConnection();
+    //! Populate the connection list combo box
+    void populateConnectionList();
+    //! Sets the actual position in connection list
+    void setConnectionListPosition();
+    //! Sets the actual position in types connection list
+    void setConnectionTypeListPosition();
+    //! Sets the selected connection type
+    void setSelectedConnectionType();
+    //! Sets the selected connection
+    void setSelectedConnection();
+
+    void on_buttonSelectSrc_clicked();
+    void on_radioSrcFile_toggled( bool checked );
+    void on_radioSrcDirectory_toggled( bool checked );
+    void on_radioSrcDatabase_toggled( bool checked );
+    void on_radioSrcProtocol_toggled( bool checked );
+    void on_btnNew_clicked();
+    void on_btnEdit_clicked();
+    void on_btnDelete_clicked();
+    void on_cmbDatabaseTypes_currentIndexChanged( const QString &text );
+    void on_cmbConnections_currentIndexChanged( const QString &text );
+    void on_buttonBox_helpRequested() { QgsHelp::openHelp( QStringLiteral( "working_with_vector/supported_data.html#loading-a-layer-from-a-file" ) ); }
```

Comment:
Could you replace this link by `managing_data_source/opening_data.html#loading-a-layer-from-a-file`